src/main/kotlin/org/openrndr/panel/elements/PlanarPad.kt

```diff
@@ -0,0 +1,198 @@
+package org.openrndr.panel.elements
+
+import io.reactivex.subjects.PublishSubject
+import org.openrndr.*
+import org.openrndr.color.ColorRGBa
+import org.openrndr.draw.Drawer
+import org.openrndr.math.Vector2
+import org.openrndr.math.clamp
+import org.openrndr.math.map
+import org.openrndr.panel.style.Color
+import org.openrndr.panel.style.color
+import org.openrndr.text.Writer
+import kotlin.math.pow
+import kotlin.math.round
+
+
+class PlanarPad : Element(ElementType("planar-pad")) {
+    var minX = -1.0
+    var minY = -1.0
+    var maxX = 1.0
+    var maxY = 1.0
+
+    // A smaller number so it doesn't clutter the UI by default
+    var precision = 1
+
+
+    // The value is derived from the normalized value...
+    var normalizedValue = Vector2(-0.0, 0.0)
+
+    var value: Vector2
+        get() = Vector2(
+                map(-1.0, 1.0, minX, maxX, normalizedValue.x).round(precision),
+                map(-1.0, 1.0, minY, maxY, normalizedValue.y).round(precision)
+        )
+        set(newValue) {
+            normalizedValue = Vector2(
+                map(minX, maxX, -1.0, 1.0, newValue.x),
+                map(minY, maxY, -1.0, 1.0, newValue.y)
+            )
+        }
+
+    init {
+        mouse.clicked.subscribe {
+            it.cancelPropagation()
+            pick(it)
+        }
+
+        mouse.dragged.subscribe {
+            it.cancelPropagation()
+            pick(it)
+        }
+
+        mouse.pressed.subscribe {
+            it.cancelPropagation()
+        }
+
+        keyboard.pressed.subscribe { handleKeyEvent(it) }
+        keyboard.repeated.subscribe { handleKeyEvent(it) }
+    }
+
+    class ValueChangedEvent(val source: PlanarPad,
+                            val oldValue: Vector2,
+                            val newValue: Vector2)
+
+
+    val events = Events()
+
+    class Events {
+        val valueChanged = PublishSubject.create<ValueChangedEvent>()
+    }
+
+
+    private fun handleKeyEvent(keyEvent: KeyEvent) {
+        val delta = 10.0.pow(-precision + 2) // +2, otherwise it's way too freaking small of a change
```

Comment:
`delta` needs to be configurable, if the interval between min/max is narrow this leads to the cursor going out of the graph.